cnd/src/db/tables.rs

```diff
@@ -0,0 +1,733 @@
+use crate::{
+    db::{
+        schema::{
+            address_hints, halight_swap_details, herc20_swap_details, secret_hashes,
+            shared_swap_ids, swaps,
+        },
+        wrapper_types::{
+            custom_sql_types::{Text, U32},
+            BitcoinNetwork, Erc20Amount, EthereumAddress, Satoshis,
+        },
+        Error, Sqlite,
+    },
+    lightning,
+    swap_protocols::{self, rfc003, HashFunction, LocalSwapId, Role},
+};
+use diesel::{self, prelude::*, RunQueryDsl};
+use libp2p::{Multiaddr, PeerId};
+
+// TODO: Audit 'pub' scope on all structs.
+
+#[derive(Identifiable, Queryable, PartialEq, Debug)]
+#[table_name = "swaps"]
+pub struct Swap {
+    id: i32,
+    local_swap_id: Text<LocalSwapId>,
+    role: Text<Role>,
+    counterparty_peer_id: Text<PeerId>,
+}
+
+impl From<Swap> for ISwap {
+    fn from(swap: Swap) -> Self {
+        ISwap {
+            local_swap_id: swap.local_swap_id,
+            role: swap.role,
+            counterparty_peer_id: swap.counterparty_peer_id,
+        }
+    }
+}
+
+#[derive(Insertable, Debug, Clone)]
+#[table_name = "swaps"]
+pub struct ISwap {
+    local_swap_id: Text<LocalSwapId>,
+    role: Text<Role>,
+    counterparty_peer_id: Text<PeerId>,
+}
+
+impl ISwap {
+    pub fn new(swap_id: LocalSwapId, counterparty: PeerId, role: Role) -> Self {
+        ISwap {
+            local_swap_id: Text(swap_id),
+            role: Text(role),
+            counterparty_peer_id: Text(counterparty),
+        }
+    }
+}
+
+#[derive(Associations, Clone, Copy, Debug, Identifiable, Queryable, PartialEq)]
+#[belongs_to(Swap)]
+#[table_name = "secret_hashes"]
+pub struct SecretHash {
+    id: i32,
+    swap_id: i32,
+    secret_hash: Text<rfc003::SecretHash>,
+}
+
+#[derive(Insertable, Debug, Clone, Copy)]
+#[table_name = "secret_hashes"]
+pub struct ISecretHash {
+    swap_id: i32,
+    secret_hash: Text<rfc003::SecretHash>,
+}
+
+// At the moment the address hint is just associated with the swap.
```

Comment:
See my comment on the table definition. I think it should be related to the peer id.

Reply:
All this database code is going to need iterating on, at the moment I'm optimising for the simplest thing that works. FK'ing on swap_id as is done for all the other tables makes the code the same so makes it easier to write quickly. Can we leave this to be improved upon later?

Reply:
The simplest thing IMO is not associating the address_hint with anything at all but just storing key-value pairs of peer-id to address, no?

Reply:
Yes, that opens us up to clobbering address hints but that is probably not an issue.  Will update with this suggestion.

Reply:
This is implemented now.